linux/network/proxy.sls

```diff
@@ -1,25 +1,85 @@
 {%- from "linux/map.jinja" import network with context %}
 {%- if network.enabled %}
-
 {%- if grains.os_family == 'Debian' %}
 
-{%- if network.proxy.host == 'none' %}
-
-/etc/profile.d/proxy.sh:
-  file.absent
-
-/etc/apt/apt.conf.d/95proxies:
-  file.absent
-
+{%- if network.proxy.get('enabled', 'True') or network.proxy.host != 'none' %}
+{%- if network.proxy.host %}
+{%- set fallback_default = network.proxy.host if not network.proxy.get('port', None) else network.proxy.host~":"~network.proxy.port %}
 {%- else %}
-
+{%- set fallback_default=None %}
+{%- endif %}
+#APT
+{%- if network.proxy.apt.enabled %}
 /etc/apt/apt.conf.d/95proxies:
   file.managed:
   - template: jinja
-  - source: salt://linux/files/95proxies
+  - source: salt://linux/files/95proxies.apt.conf
+  - defaults:
+      https: {{ network.proxy.apt.get('https', None) or network.proxy.get('https', None)  | default(fallback_default, true) }}
+      http: {{ network.proxy.apt.get('http', None) or network.proxy.get('http', None)  | default(fallback_default, true) }}
+      ftp: {{ network.proxy.apt.get('ftp', None) or network.proxy.get('ftp', None)  | default(fallback_default, true) }}
+{%- else %}
+/etc/apt/apt.conf.d/95proxies:
+  file.absent
+{%- endif %}
 
+# SYSTEM LEVEL VARIABLES
+{%- if network.proxy.system.enabled or d or network.proxy.noproxy.enabled %}
+# on shared profiles (affect any .sh execution), keep this for backward compatibility
+/etc/profile.d/95proxies:
+  file.managed:
+  - template: jinja
+  - source: salt://linux/files/95proxies.profile.d
+  - defaults:
+      noproxy: {{ network.proxy.system.get('noproxy', None) or network.proxy.get('noproxy', None) }}
+      https: {{ network.proxy.system.get('https', None) or network.proxy.get('https', None)  | default(fallback_default, true) }}
+      http: {{ network.proxy.system.get('http', None) or network.proxy.get('http', None)  | default(fallback_default, true) }}
+      ftp: {{ network.proxy.system.get('ftp', None) or network.proxy.get('ftp', None)  | default(fallback_default, true) }}
+# on environment (realy system wide, but keem in mind no variable expansion)
+/etc/environment_add_proxies:
```

Comment:
Resource name is confusing, is path but doesn't correspond with file that's being modified.
Use `linux_system_environment_proxies` or something